src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -94,10 +137,10 @@ abstract class AbstractLdapNamingEnumeration<T extends NameClassPair>
                 refEx = answer.refEx;
             }
 
+            this.enumCtx = new EnumCtx(homeCtx, answer, homeCtx.clnt);
             // Ensures that context won't get closed from underneath us
-            this.homeCtx = homeCtx;
-            homeCtx.incEnumCount();
-            enumClnt = homeCtx.clnt; // remember
+            this.enumCtx.homeCtx.incEnumCount();
```

Comment:
@bchristi-git Would it make sense to increment this in the EnumCtx ctor instead?